cpp/include/cudf/stream_compaction.hpp

```diff
@@ -214,7 +214,38 @@ enum class duplicate_keep_option {
 };
 
 /**
- * @brief Create a new table without duplicate rows
+ * @brief Eliminates all except the row specified by `keep` from every consecutive group of
+ * equivalent rows.
+ *
+ * Given an `input` table_view, one row from a group of equivalent elements is copied to
+ * output table depending on the value of @p keep:
+ * - KEEP_FIRST: only the first of a sequence of duplicate rows is copied
+ * - KEEP_LAST: only the last of a sequence of duplicate rows is copied
+ * - KEEP_NONE: no duplicate rows are copied
+ *
+ * @throws cudf::logic_error if The `input` row size mismatches with `keys`.
+ *
+ * @param[in] input           input table_view to copy only unique rows
+ * @param[in] keys            vector of indices representing key columns from `input`
+ * @param[in] keep            keep first entry, last entry, or no entries if duplicates found
+ * @param[in] nulls_equal     flag to denote nulls are equal if null_equality::EQUAL, nulls are not
+ *                            equal if null_equality::UNEQUAL
+ * @param[in] mr              Device memory resource used to allocate the returned table's device
+ * memory
+ *
+ * @return Table with unique rows from each sequence of equivalent rows as per specified `keep`.
+ */
+std::unique_ptr<table> drop_duplicates(
+  table_view const& input,
+  std::vector<size_type> const& keys,
+  duplicate_keep_option keep,
+  null_equality nulls_equal           = null_equality::EQUAL,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Create a new table without duplicate rows.
```

Comment:
If you don't want sorting, I think a simple solution to this is to use gathering:
 * Add a gather map column (sequence of `[0, 1, 2, ...]`) to the input table.
 * After calling to `drop_duplicates`, create another gather map (sequence of `[0, 1, 2, ...]`), then black out (set `-1`) the positions which have been removed from the previous step. The second gather map here will contain an increasing sequence of the unique rows or `-1`.
 * Gather the input using the second step gather map.

Reply:
@ttnghia Are you referring to something like [this](https://github.com/rapidsai/cudf/pull/10030/files#diff-3f6f871f1d900cb983ffe17c20defbcc0ef032f0f894fbc8126babf7a62ff1afR124-R142)?

Reply:
I described in the original issue that if users want the old behavior of `drop_duplicates`, they should sort before calling it.

That said, I think it would be easier for everyone to introduce `unordered_drop_duplicates` first without changing the behavior of `drop_duplicates`.

Reply:
`std::unique`-like APIs removed.